src/renderers.ts

```diff
@@ -66,19 +66,19 @@ export const rendererRegistryEntries: JsonFormsRendererRegistryEntry[] = [
     renderer: withJsonFormsLabelProps(AlertControl),
   },
   {
-    tester: rankWith(2, isNumericControl),
-    renderer: withJsonFormsControlProps(NumericControl),
-  },
-  {
-    tester: rankWith(2, isNumericControl),
-    renderer: withJsonFormsControlProps(NumericControl),
-  },
-  {
-    tester: rankWith(2, isNumericControl),
+    tester: rankWith(2, or(isNumberControl, isIntegerControl)),
     renderer: withJsonFormsControlProps(NumericControl),
   },
   {
-    tester: rankWith(3, isNumericSliderControl),
+    tester: rankWith(
+      3,
+      and(
+        or(isNumberControl, isIntegerControl),
+        schemaMatches((schema: JsonSchema) => {
+          return schema.minimum !== undefined && schema.maximum !== undefined
+        }),
+      ),
+    ),
```

Comment:
This is short enough that I decided to just move it in here with everything else